mne/source_estimate.py

```diff
@@ -335,18 +339,39 @@ def read_source_estimate(fname, subject=None):
 
     if ftype == 'volume':
         stc = VolSourceEstimate(**kwargs)
+    elif ftype == 'h5' and kwargs['data'].ndim == 3:
+        stc = VectorSourceEstimate(**kwargs)
     else:
         stc = SourceEstimate(**kwargs)
 
     return stc
 
 
-def _make_stc(data, vertices, tmin=None, tstep=None, subject=None):
-    """Generate a surface, volume or mixed source estimate."""
+def _make_stc(data, vertices, tmin=None, tstep=None, subject=None,
+              vector=False, source_nn=None):
+    """Generate a surface, vector-surface, volume or mixed source estimate."""
+
     if isinstance(vertices, list) and len(vertices) == 2:
         # make a surface source estimate
-        stc = SourceEstimate(data, vertices=vertices, tmin=tmin, tstep=tstep,
-                             subject=subject)
+        n_vertices = len(vertices[0]) + len(vertices[1])
+        if vector:
+            if source_nn is not None:
```

Comment:
Ah, this is a leftover from when `fixed=True` was still reported. Data should enter the `_make_stc` function in surface coordinates, and should always be stored in the `VectorSourceEstimate` object in MRI coordinates. This rotation and reshaping of the data is a piece of data mangling that needs to done at the end of each `apply_inverse_*` function. I think the current location for this data mangling is fine.